### ISSUE TITLE
`image` and `gallery` block drawers not fully closing when navigating away

### ISSUE THREAD

**Issue by silllli** [k-next/kirby#6778]
## Description
When accessing a file from an `image` or `gallery` block drawer, leftovers of the drawer remain visible.

**Expected behavior**  
See the file view without bugs.

**Screenshots**  
![Bildschirmfoto 2024-10-29 um 15 00 55](https://github.com/user-attachments/assets/97668a40-cf01-4281-9c9c-d3adb4834494)

## To reproduce

1. Open an instant online demo
2. Go to the `Blog` example
3. Open the first article
4. Open the image block drawer
5. Click on the file
6. See the broken file view

## Your setup
**Kirby Version**  
4.4.1

**Console output**  
Nothing unusual

**Your system (please complete the following information)**
 - Device: MacBook Pro 16" (2019)
 - OS: macOS 15.1
 - Browser: Safari
 - Version: Version 18.1 (20619.2.8.11.10)

**Comment by silllli** [comment]
It also happens when clicking on a user of a `users` field inside an open drawer.

**Comment by afbora** [comment]
I think we need to fix with `beforeunload` event. Maybe we can solve this with the `beforeunload` listeners newly added to v5. I've tried to fix it in v4 and v5 but couldn't achieve it.

Something like that:

````js
mounted() {
    this.$events.on("beforeunload", this.close);
},
destroyed() {
    this.$events.off("beforeunload", this.close);
},
````

**Comment by distantnative** [comment]
@afbora Have you tried that? I don't think the get called when navigating around the Panel, just leaving the Panel/closing the browser.

**Comment by afbora** [comment]
@distantnative Yes, I tried but I thought it would work during page transitions but it didn't. Maybe I couldn't do it.

**Comment by distantnative** [comment]
Related to https://github.com/getkirby/kirby/issues/6030

**Comment by bastianallgeier** [comment]
This issue has been fixed in the meantime.
